src/app/core/mods/zaapSearchFilter/zaapSearchFilter.ts

```diff
@@ -0,0 +1,116 @@
+import { Logger } from "app/core/electron/logger.helper";
+
+import { Mod } from "../mod";
+
+
+export class ZaapSearchFilter extends Mod {
+
+    private styleTag: HTMLDivElement;
+    private zaapSearchContainer: HTMLDivElement;
+    private zaapSearchInput : HTMLInputElement;
+    private inputPlaceholder: string;
+    
+
+    startMod(): void {
+        this.params = this.settings.option.vip.general.zaapsearchfilter;
+        this.inputPlaceholder = this.translate.instant('app.option.vip.zaapsearchfilter.placeholder');
+
+        if(this.params){
+            Logger.info('- enable ZaapSearchFilter');
+
+            this.on(this.wGame.dofus.connectionManager, 'ZaapListMessage', (msg: any) => {
+                this.createSearchFilter();
+            });
+    
+            this.on(this.wGame.connectionManager, 'LeaveDialogMessage', (msg: any) => {
+                this.resetSearchFilter();
+            });
+        }
+    }
+
+
+    private createSearchFilter(): void{
+        this.injectInputInDom();
+
+        this.zaapSearchInput.addEventListener("change", () => {
+            let zaapWanted = this.zaapSearchInput.value;
+            let zaapList = this.wGame.document.getElementsByClassName('zaapBodyHeight__custom')[0]
```

Comment:
**Suggestion**

Consider using a finer selector that a class on the document, as to avoid any selecting errors